### ISSUE TITLE
`table.sort` misbehaving

### ISSUE THREAD

**Issue by Ocawesome101** [MCJack123/craftos2#274]
**Describe the bug**
CraftOS-PC's `table.sort` behavior does not match CCEmuX.  This might also be a misbehaving string length operator. 

**To Reproduce**
1. Launch Recrafted on both CraftOS-PC and CCEmuX.
2. Run `bedit /rom/apis/window.lua`
3. Observe that the `into_buffer` function name is truncated to `in` on CraftOS-PC, and not on CCEmuX.

**Expected behavior**
Behavior should be consistent across both platforms.  CCEmuX's behavior matches PUC Lua's. 

The relevant `table.sort` call is [here](https://github.com/Ocawesome101/recrafted/blob/primary/data/computercraft/lua/rom/modules/main/edit/syntax.lua#L99-L103).

**Screenshots**
![Screenshot_20220906_130114](https://user-images.githubusercontent.com/50844998/188696459-76a4e257-7e78-49df-93ba-a88ac39d7964.png)

**Environment (please complete the following information):**
 - OS: Arch Linux
 - OS Version: _all of them_
 - CraftOS-PC Version: 2.7.1
 - Compiled from source? Yes (AUR)

**Comment by MCJack123** [comment]
If you can, I would highly appreciate a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example), as this'll make debugging this issue much easier for me.

**Comment by Ocawesome101** [comment]
this should do it
```lua
-- sorting weirdness

local tab = {
  {
    value = "into_buffer",
    class = "lol"
  },
  {
    value = "in",
    class = "foo"
  }
}

table.sort(tab, function(a, b)
  return #a.value > #b.value
    or b.class == "lol"
end)

for i=1, #tab, 1 do
  print(tab[i].value)
end
```

Interestingly, PUC Lua (5.1 and 5.4) displays the same behavior as CraftOS-PC - I wonder if CCEmuX is the odd one out here and my code is in fact incorrect.

**Comment by MCJack123** [comment]
In general, you should avoid using equality comparisons inside the sort function, or at least make sure you check the same thing on both sides and return correctly - otherwise, the search is unstable, and the result is undefined. In this example, if `#a.value <= #b.value` and `b.class == "lol"`, the function returns `true`; however, flipping the objects will give `#a.value > #b.value`, resulting in `true` again. This could make the function infinitely cycle, at which point `table.sort` would report that as an error.

TL;DR this is undefined behavior, and I don't plan on rewriting the entire sorting function to match such undefined behavior, especially since the UB may include (rightly) erroring.
